port/scorecard/resouce.go

```diff
@@ -0,0 +1,152 @@
+package scorecard
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/port-labs/terraform-provider-port-labs/internal/cli"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ScorecardResource{}
+var _ resource.ResourceWithImportState = &ScorecardResource{}
+
+func NewScorecardResource() resource.Resource {
+	return &ScorecardResource{}
+}
+
+type ScorecardResource struct {
+	portClient *cli.PortClient
+}
+
+func (r *ScorecardResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_scorecard"
+}
+
+func (r *ScorecardResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.portClient = req.ProviderData.(*cli.PortClient)
```

Comment:
whats the `ProviderData.()` syntax mean?

Reply:
https://developer.hashicorp.com/terraform/plugin/framework/resources/configure